nixos-manual: Include all modules in documentation generation

arianvp opened the pull request:
Previously,  `man 5 configuration.nix` would only show
options of modules in the `baseModules` set, which
consists only of the list of modules in `nixos/modules/module-list.nix`

however, with this change, any extra module that is included
in your `configuration.nix` file by the means of

```
  imports = [ ./rootfs.nix ]
```

will also be included the generated documentation.

Example output:

```
OPTIONS
       You can use the following options in configuration.nix.

       arian.rootfs.device
           The device that contains the btrfs volume

           Type: string

           Declared by:
               /home/arian/Projects/nixos-stuff/computers/t430s/rootfs.nix

       boot.enableContainers
           Whether to enable support for nixos containers.

           Type: boolean

           Default: true

           Declared by:
               <nixpkgs/nixos/modules/virtualisation/containers.nix>
```

###### Motivation for this change

I found it useful for my machine to be self-documenting.  It also means
that if you import modules like `gce.nix` or `azure.nix` which aren't in
`module-list.nix`,  documentation is still generated for these modules

cc:  https://github.com/NixOS/nixpkgs/issues/11499

###### Things to do

- [ ] I have no idea if this patch impacts the way the documentation for https://nixos.org/nixos
is generated. It might.  We should check this.


---

arianvp commented:
/cc @oxij

arianvp commented:
Ah apparently using `<nixos-config>` doesn't work here. Alas

oxij commented:
I like the idea, but

- I think you can just add `modules` argument to the top of the whole expression and get the whole list of modules loaded by the current configuration.
- Hence, I expected this to simply add `++ optionals optionName modules` thing after `baseModules`, not to rewrite the whole attribute.

E.g. why did you go from `evalModules` back to `eval-config.nix` and remove all the stuff the old version did with `localSystem`?

arianvp commented:
Just my unfamiliarity. I somehow was convinced that the modules argument was not injected into each module by `evalModules` so that's why I evalled `<nixos-config>` instead.   If `modules` is indeed passed as a parameter than that is indeed a more elegant solution and will also not break CI. I'll investigate tomorrow. Thanks for the feedback!

arianvp committed:
nixos-manual: Include all modules in documentation generation

Previously,  `man 5 configuration.nix` would only show
options of modules in the `baseModules` set, which
consists only of the list of modules in `nixos/modules/module-list.nix`

however, with this change, any extra module that is included
in your `configuration.nix` file by the means of

```
  imports = [ ./rootfs.nix ]
```

will also be included the generated documentation.

Example output:

```
OPTIONS
       You can use the following options in configuration.nix.

       arian.rootfs.device
           The device that contains the btrfs volume

           Type: string

           Declared by:
               /home/arian/Projects/nixos-stuff/computers/t430s/rootfs.nix

       boot.enableContainers
           Whether to enable support for nixos containers.

           Type: boolean

           Default: true

           Declared by:
               <nixpkgs/nixos/modules/virtualisation/containers.nix>
```

I found it useful for my machine to be self-documenting.  It also means
that if you import modules like `gce.nix` or `azure.nix` which aren't in
`module-list.nix`,  documentation is still generated for these modules

cc:  https://github.com/NixOS/nixpkgs/issues/11499

- [ ] I have no idea if this patch impacts the way the documentation for https://nixos.org/nixos
is generated. It might.  We should check this.

---

arianvp commented:
I haved used your suggestions! Seems to build now on CI!
